gtda/utils/metaestimators.py

```diff
@@ -0,0 +1,137 @@
+"""Meta-estimators."""
+# License: GNU AGPLv3
+
+from functools import reduce
+from operator import and_
+
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.base import clone
+from sklearn.utils.metaestimators import if_delegate_has_method
+
+from gtda.utils import check_collection
+
+
+class ForEachInput(BaseEstimator, TransformerMixin):
+    """Meta-transformer for applying a fit-transformer to each input in a
+    collection.
+
+    If `transformer` possesses a ``fit_transform`` method,
+    ``ForEachInput(transformer)`` also possesses a :meth:`fit_transform` method
+    which, on each entry in its input ``X``, fit-transforms a clone of
+    `transformer`. A collection (list or ndarray) of outputs is returned.
+
+    Parameters
+    ----------
+    transformer : object
+        The fit-transformer instance from which the transformer acting on
+        collections is built. Should implement ``fit_transform``.
+
+    n_jobs : int or None, optional, default: ``None``
```

Comment:
In #490 , I proposed to call this `outer_n_jobs` (and analogously for the next parameter). Here, I am proposing to remove the "outer" to have consistency with `make_mapper_pipeline`.

Reply:
I agree, this is also consistent with other `sklear` metaestimators such as `GridSearchCV`.